### PR TITLE
Feature material list view

### PR DESCRIPTION
**Pull request by cHgcH**

### CODE REVIEW

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java [hunk 1]
```diff
@@ -29,6 +36,10 @@
     @Inject
     SecretPrompter secretPrompter;
 
+    RecyclerView recyclerView;
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java]: Should be private.

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java [hunk 2]
```diff
@@ -29,6 +36,10 @@
...
+    NotesAdapter adapter;
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java]: Should be private.

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java [hunk 3]
```diff
@@ -29,6 +36,10 @@
...
+    SwipeToAction swipeToAction;
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java]: Should be private.

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/NotesAdapter.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+public class NotesAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private List<Note> items;
+
+
+    /** References to the views for each data item **/
+    public class NoteViewHolder extends SwipeToAction.ViewHolder<Note> {
+        public TextView titleView;
+        public TextView authorView;
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/NotesAdapter.java]: Rename from authorView to contentView.

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/NotesAdapter.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+        public NoteViewHolder(View v) {
+            super(v);
+
+            titleView = (TextView) v.findViewById(R.id.title);
+            authorView = (TextView) v.findViewById(R.id.author);
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/NotesAdapter.java]: change id from author to content.

File: CyFile-Android/app/src/main/res/layout/item_view.xml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        android:background="@color/accent"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/note_height">
+
+        <ImageView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignParentLeft="true"
+            android:layout_centerVertical="true"
+            android:layout_marginLeft="20dp"
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/res/layout/item_view.xml]: Convert hardcoded value into variable from dimen.xml

File: CyFile-Android/app/src/main/res/layout/item_view.xml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        android:background="@color/colorDeleteButton"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/note_height">
+
+        <ImageView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignParentRight="true"
+            android:layout_centerVertical="true"
+            android:layout_marginRight="20dp"
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/res/layout/item_view.xml]: Convert from hardcoded to dimen.xml variable.

File: CyFile-Android/app/src/main/res/layout/item_view.xml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+        <TextView
+            android:id="@+id/title"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_toEndOf="@id/note_image"
+            android:layout_marginTop="@dimen/minimal_margin_dp"
+            android:layout_marginStart="@dimen/small_margin_dp"
+            android:singleLine="true"
+            android:textSize="16dp"
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/res/layout/item_view.xml]: Convert from hardcoded to dimen.xml

File: CyFile-Android/app/src/main/res/layout/item_view.xml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+            android:textStyle="bold" />
+
+        <TextView
+            android:id="@+id/author"
+            android:layout_marginStart="@dimen/small_margin_dp"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignLeft="@+id/title"
+            android:layout_below="@+id/title"
+            android:textSize="14dp" />
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/res/layout/item_view.xml]: convert from hardcoded to dimen.xml

File: CyFile-Android/build.gradle [hunk 10]
```diff
@@ -23,6 +23,7 @@ buildscript {
         classpath 'com.android.tools.build:gradle:3.1.2'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
 
+
```
Comment (Cobrijani) [inline CyFile-Android/build.gradle]: remove blank line

File: CyFile-Android/app/src/main/res/layout/item_view.xml [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    </RelativeLayout>
+
+
+    <!-- just a divider -->
+    <View android:background="@color/divider"
+        android:layout_width="fill_parent"
+        android:layout_height="0.5dp" />
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/res/layout/item_view.xml]: convert from hardcoded value to dimen.xml

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java [hunk 12]
```diff
@@ -39,29 +50,47 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         secretPrompter.promptSecret();
     }
 
+    protected void initializeNoteList()
```
Comment (Cobrijani) [inline CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/MainActivity.java]: Constructing the view should be done only in onCreate method.

**Comment by cHgcH** [comment]
Updated the code and fixed the issues

**Pull request merged**